x-pack/test/detection_engine_api_integration/security_and_spaces/rule_execution_logic/new_terms.ts

```diff
@@ -622,7 +640,10 @@ export default ({ getService }: FtrProviderContext) => {
           query: { match: { id: 'first_doc' } },
           index: 'new_terms',
           fields: [AGG_FIELD_NAME],
-          runtimeMappings: getNewTermsRuntimeMappings(['user.name', 'user.id']),
+          runtimeMappings: getNewTermsRuntimeMappings(['user.name', 'user.id'], {
+            'user.name': { 'user-0': true },
+            'user.id': { '0': true },
```

Comment:
Could you add a test to verify that numeric fields like `user.id` are encoded the same way by `transformBucketsToValues` as they are by the runtime field? A full rule preview execution searching for `first_doc` seems like the most robust way to check, but extracting the `expectedEncodedValues` constant and using it in both this test and a test of `transformBucketsToValues` could also work.

Since we rely on the encoding of values in Kibana to match up with the encoding in Elasticsearch, we should be extra careful to verify that they match up for both string and numeric source data types.

Reply:
Values in painless script and in Kibana are converted to string before encoding.

Plus, there is a set of tests for various types: as this one https://github.com/elastic/kibana/blob/main/x-pack/test/detection_engine_api_integration/security_and_spaces/rule_execution_logic/new_terms.ts#L330-L344
It basically plays role as integration test between `transformBucketsToValues` and runtime script, as if encoding differs, there wouldn't be generated alerts. So, I think we covered here for this and similar use cases